### ISSUE TITLE
Gradient clipping support

### ISSUE THREAD

**Issue by petrbel** [iterait/emloop-tensorflow#6]
BaseModel._create_train_ops should be configurable to perform gradient clipping. This will allow us to implement and train more advanced models such as seq2seq etc.
